CONTRIBUTE.md

```diff
@@ -0,0 +1,333 @@
+- Spread syntax (`myFunction(...iterableObj);`)
+
+## Providing Context
+
+Where possible it may be useful to provide examples showing how a method could be used in the context of a function or class. It may not always be possible to provide a succinct example and a series of `console.log`s can also be a valid way of illustrating an example.
+
+## Tangible examples
+
+It's nice to be able to illustrate a programming concept using real or imaginary world examples, if possible using the same theme throughout a specific topic.
+
+For example:
+
+```
+function 1(disposition) {
+  this.disposition = disposition;
+}
+
+var handler1 = {
+  construct: function(target, args) {
+    console.log("1 constructor called");
+    // expected output: " constructor called"
+
+    return new target(...args);
+  }
+};
+
+var proxy1 = new Proxy(1, handler1);
+
+console.log(new proxy1("fierce").disposition);
+// expected output: "fierce"
+```
+
+
+## Dealing with Errors
+
+Writing code that will throw an error can be useful way to illustrate an example. However an uncaught error will halt execution immediately and will not display the results of any `console.log` statements in the example.
+
+If you wish to use an error to illustrate a method you could wrap it in a `try` `catch` clause, for example:
+
+```
+try {
+  Intl.getCanonicalLocales('EN_US');
+} catch (err) {
+  console.log(err);
+  // expected output: RangeError: invalid language tag: EN_US
+}
+```
+
+## Format
+
+The aim is to try and fit examples into 12 lines of code, if it's not possible to create a useful example in 12 lines, it's fine to use more lines. We currently have examples of up to 25 lines.
+
+### Semi-colons
+
+There are valid arguments for and against using semi-colons. We use them.
+
+### Line Spacing
+
+To fit the space available we try and keep examples as compact as possible, using blank lines sparingly to increase readability.  
+
+One place we use a blank line is immediately preceding a `console.log` statement unless it is the first line in a block.
+
+For example:
+
+```
+construct: function(target, args) {
+  console.log("1 constructor called");
+  // expected output: " constructor called"
+
+  return new target(...args);
+}
+```
+and
+
+```
+var proxy1 = new Proxy(1, handler1);
+
+console.log(new proxy1("fierce").disposition);
+// expected output: "fierce"
+```
+
+### Indentation
+
+In order to keep things as concise as reasonably possible we indent with two spaces.
+
+### Instantiating an Object
+
+Most objects are instantiated in the normal way, for example:
+```
+var date1 = new Date();
+```
+
+The exception being `Object` in which case we instantiate like so:
+
+```
+var object1 = {};
+```
+
+### Instantiating an Array
+
+We instantiate arrays in the following way:
+
+```
+var array1 = [2, 5, 7, 9];
+```
+
+Note the spacing after the commas.
+
+### Passing parameters
+
+We space after commas when passing parameters to a function, but no space padding of parenthesis:
```

Comment:
Nit: "We use a space after commas when passing parameters to a function, but do not pad the parenthesis: i.e.
```
calcAngle(8, 10);
```
but not,

```
calcAngle( 8, 10 );
```
"

Reply:
👍